arbor/morph/region.cpp

```diff
@@ -306,9 +307,329 @@ std::ostream& operator<<(std::ostream& o, const all_& t) {
     return o << "(all)";
 }
 
+// Region with all segments distal from another region
 
-// Named region.
+struct distal_interval_ {
+    locset start;
+    double distance; //um
+};
+
+region distal_interval(locset start, double distance) {
+    return region(distal_interval_{start, distance});
+}
+
+mcable_list thingify_(const distal_interval_& reg, const mprovider& p) {
+    const auto& m = p.morphology();
+    const auto& e = p.embedding();
+
+    std::vector<mcable> L;
+
+    auto start = thingify(reg.start, p);
+    auto distance = reg.distance;
+
+    struct branch_interval {
+        msize_t bid;
+        double distance;
+    };
+
+    for (auto c: start) {
+        std::stack<branch_interval> branches_reached;
+        bool first_branch = true;
+
+        // Transform end point of branch into the start point of its parent
+        if (c.pos == 0) {
+            c = {m.branch_parent(c.branch),1};
+        }
+
+        // if we're starting at the end of a branch, start traversal with its children
+        if (c.pos < 1) {
```

Comment:
I think this is incorrect in the case where distance is zero: here, we'd expect a location (bid, 1) in start to give a cable (bid, 1, 1) in the result. (This would be fixed up by a `remove_cover` or `remove_covered_points` (?) call at the end, though, which I think we need anyway.)